audb/core/load.py

```diff
@@ -1037,6 +1037,129 @@ def load(
     return db
 
 
+def load_attachments(
+        name: str,
+        attachments: typing.Union[str, typing.Sequence[str]],
+        *,
+        version: str = None,
+        cache_root: str = None,
+        num_workers: typing.Optional[int] = 1,
+        timeout: float = -1,
+        verbose: bool = True,
+) -> audformat.Database:
+    r"""Load attachment(s) of database.
+
+    Args:
+        name: name of database
+        attachments: attachment ID
+            or sequence of attachment IDs
+            to load
+        version: version of database
+        cache_root: cache folder where databases are stored.
+            If not set :meth:`audb.default_cache_root` is used
+        num_workers: number of parallel jobs or 1 for sequential
+            processing. If ``None`` will be set to the number of
+            processors on the machine multiplied by 5
+        timeout: maximum wait time if another thread or process is already
+            accessing the database. If timeout is reached, ``None`` is
+            returned. If timeout < 0 the method will block until the
+            database can be accessed
+        verbose: show debug messages
+
+    Returns:
+        list of attachment file paths
+
+    Raises:
+        ValueError: if an attachment ID is requested
+            that is not part of the database
+
+    Examples:
+        >>> load_attachments(
+        ...     'emodb',
+        ...     [],
+        ...     version='1.3.0',
+        ...     verbose=False,
+        ... )
+        []
+
+    """
+    attachments = audeer.to_list(attachments)
+    if version is None:
+        version = latest_version(name)
+
+    db_root = database_cache_root(name, version, cache_root)
+
+    if verbose:  # pragma: no cover
+        print(f'Get:   {name} v{version}')
+        print(f'Cache: {db_root}')
+
+    deps = dependencies(
```

Comment:
I think this the problem why the test does not raise a warning: calling `dependencies()` locks `db_root` with timeout set to -1. I.e. the first thread that enters gets the lock and the second thread will wait. But when the first thread finishes it will immediately again locks `db_root` before the second thread is actually awoken. Thread one now finishes its work and only then the second thread actually passes this line of code and when it arrives in line 1127 can acquire the lock. With `load_media()` we do not have this issue because we lock the flavor folder, which is different from the folder locked by `dependencies()`. You can verify that this the problem by adding an extra sleep after `dependencies()` that blocks the first thread long enough that the second thread can wake up. Now the two threads can actually get into a race condition.